hre/src/main/java/hre/util/Notifier.scala

```diff
@@ -0,0 +1,95 @@
+package hre.util
+
+import java.io.{ByteArrayInputStream, File}
+
+import sys.process._
+import hre.lang.System.Warning
+
+import scala.collection.JavaConverters._
+import java.nio.file.{Files, Paths}
+
+object Notifier {
+  def notify(title: String, message: String): Boolean = {
+    val os = System.getProperty("os.name").toLowerCase()
+    if (os.contains("win")) {
+      notifyWindows10(title, message)
+    } else if (os.contains("mac")) {
+      notifyMacOS(title, message)
+    } else if (os.contains("nix") || os.contains("linux")) {
+      notifyLibnotify(title, message)
+    } else {
+      false
+    }
```

Comment:
this logic deviates a bit from `Configuration` (but seems slightly better), we should standardize it somewhere.

Reply:
I can merge them in a helper method in `Configuration`

Reply:
I think I duplicated the logic a few times there, sorry :D